sql/plan/prepare.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020-2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// PrepareQuery is a node that prepares the query
+type PrepareQuery struct {
+	Name  string
+	Child sql.Node
+}
+
```

Comment:
Good practice to always add the `var _ sql.Node = (*PrepareQuery)(nil)` lines to get type checking that our implementations always fully implement any interfaces we want them to. It's a great safeguard to have in the future when we inevitably need to change those interfaces.